contracts/test-utils/src/unit_test_helper.ts

```diff
@@ -0,0 +1,82 @@
+import { RevertError } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { ContractArtifact, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+import { expect } from './chai_setup';
+import { LogDecoder } from './log_decoder';
+
+type AsyncFunction<TArgs extends any[], TResult> = (...args: TArgs) => Promise<TResult>;
+
+interface ContractArtifacts {
+    [contractName: string]: ContractArtifact;
+}
+
+interface TransactionReturnData<TCallAsyncResult> {
+    result: TCallAsyncResult;
+    receipt: TransactionReceiptWithDecodedLogs;
+}
+
+interface ContractWrapperFunction<
+    TCallAsyncArgs extends any[],
+    TAwaitTransactionSuccessAsyncArgs extends any[],
+    TCallAsyncResult
+> {
+    callAsync: AsyncFunction<TCallAsyncArgs, TCallAsyncResult>;
+    sendTransactionAsync?: AsyncFunction<TAwaitTransactionSuccessAsyncArgs, string>;
+}
+
+export type TransactionExpectation<TAwaitTransactionSuccessAsyncArgs extends any[], TCallAsyncResult> = (
+    tx: Partial<TransactionReturnData<TCallAsyncResult>>,
+    // tslint:disable-next-line: trailing-comma
+    ...args: TAwaitTransactionSuccessAsyncArgs
+) => Promise<void> | void;
+
+export class BaseUnitTestHelper<TContract> {
+    protected readonly _testContract: TContract;
+    private readonly _logDecoder: LogDecoder;
+
+    protected static async _verifyContractMethodRevertErrorAsync<
+        TCallAsyncArgs extends any[],
+        TAwaitTransactionSuccessAsyncArgs extends any[],
+        TCallAsyncResult
+    >(
+        contractFunction: ContractWrapperFunction<TCallAsyncArgs, TAwaitTransactionSuccessAsyncArgs, TCallAsyncResult>,
+        expectedError: RevertError,
+        // tslint:disable-next-line: trailing-comma
+        ...args: TAwaitTransactionSuccessAsyncArgs
+    ): Promise<void> {
+        const tx = contractFunction.sendTransactionAsync
+            ? contractFunction.sendTransactionAsync(...args)
+            : contractFunction.callAsync(...((args as any) as TCallAsyncArgs));
+        await expect(tx).to.revertWith(expectedError);
```

Comment:
Optional nit: Oftentimes we will use a `return` here instead of `await` since this is at the end of the function. If possible, I think this would be a bit nicer.